dttools/src/jx_match.c

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright (C) 2016- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include <stdarg.h>
+#include <string.h>
+
+#include "jx.h"
+#include "jx_match.h"
+
+void *jx_match_null(struct jx *j) {
+	return jx_istype(j, JX_NULL) ? (void *) j : NULL;
+}
+
+int *jx_match_boolean(struct jx *j, int *v) {
+	if (jx_istype(j, JX_BOOLEAN)) {
+		if (v) {
+			*v = !!j->u.boolean_value;
+		}
+		return &j->u.boolean_value;
+	} else {
+		return NULL;
+	}
+}
+
+jx_int_t *jx_match_integer(struct jx *j, jx_int_t *v) {
+	if (jx_istype(j, JX_INTEGER)) {
+		if (v) {
+			*v = j->u.integer_value;
+		}
+		return &j->u.integer_value;
+	} else {
+		return NULL;
+	}
+}
```

Comment:
If you do not like that, you have to change your function to:

```C
const jx_int_t *jx_match_integer(struct jx *j, jx_int_t *v) { ```

as we do not want the pointer to be able to modify its value.